client/src/App.js

```diff
@@ -22,17 +22,37 @@ function App() {
   const [suggestions, setSuggestions] = useState([]);
   const [showSuggestions, setShowSuggestions] = useState(false);
   const [darkTheme, setDarkTheme] = useState(false); // State for dark theme
+  const [inputPosition, setInputPosition] = useState({ top: 0, left: 0 }); // State to track input position
 
   const messagesEndRef = useRef(null);
+  const inputRef = useRef(null);
 
   useEffect(() => {
     fetchMessages();
+    // Add event listener to listen for clicks on the document body
+    document.body.addEventListener('click', handleClickOutside);
+    return () => {
+      // Clean up event listener when component unmounts
+      document.body.removeEventListener('click', handleClickOutside);
+    };
   }, []);
 
   useEffect(() => {
     scrollToBottom();
   }, [messages]);
```

Comment:
```suggestion
  }, [messages?.length]);
```
maybe it will require the scroll action less often but still achieve the same result